test/libsolidity/semanticTests/functionTypes/selector_ternary.sol

```diff
@@ -0,0 +1,10 @@
+contract C {
+	function f() public {}
+	function g() public {}
+	function h(bool c) public returns (bytes4) {
+		return (c ? this.f : this.g).selector;
```

Comment:
I see that this PR does not cover cases that do not involve ternary operator. For example the one where the function pointer is the result of a function call.

```
contract A {
    function f() public {}
}

contract C {
    uint public i = 0;
    A[] a = new A[](3);

    function getContract() public returns (A) {
        return a[i++];
    }

    function g() public returns (bytes4) {
        return getContract().f.selector;
    }
}
// ----
// i() -> 0
// g() -> 0x26121ff000000000000000000000000000000000000000000000000000000000
// i() -> 1
```

Reply:
But that's perfectly fine, what should we do about that?
https://github.com/ethereum/solidity/pull/14366 will ensure that the function call expression is actually properly evaluated. Besides that the return type of the function is fully determined by the contract type.
I don't see a problem here.

Are the types involved actually the same? I could imagine that ``getContract().f`` has an external function type anyways (and we'll actually put the contract address on stack first (for ``getContract()``), then enriching it with a selector (yielding ``getContract().f``) and then masking out the selector again (yielding ``getContract().f.selector``)). 
In that case even ternary operators like ``(getContract().f : getContract().g).selector`` should be fine and wouldn't be affected by the PR. In case they, these expressions *do* get a ``FunctionType::Kind::Declaration`` kind after all, though, the PR should just also cover it.

So it makes sense to double-check this, but I don't see a problem here per se so far.

Reply:
Ah wait, somehow I forgot about #14366 :) I haven't reviewed it yet but looks like it does cover it.

It's a bit hard to keep track of all of it :) Maybe I should have just quickly gone over all PRs first instead of doing them in sequence and digging deep into each one.

Reply:
Yeah, that's what I meant earlier: they'll only fix everything together - maybe I shouldn't even have split them :-).

Reply:
Splitting is fine in general. Smaller PRs are better. In this case it's my fault for not checking that other PR before posting this.

It's just that there are too many cases. A monolithic PR would make it easier in some ways but harder in others.

Reply:
Easier than cross referencing 3 different PRs to make sure everything is covered though...

Reply:
That's true, although in this case, Daniel's PRs actually have fairly small changes - so having a single PR with all the tests would be nice. In any case, all have been somewhat reviewed so far, so no point in opening another PR.

Reply:
I don't really consider these changes that small. Each one changes only a few lines, but thinking through all the consequences of those changes still takes quite a bit of effort for me.